src/query.rs

```diff
@@ -257,7 +254,7 @@ pub struct Witness {
     /// Notice how this is not a compilable example, but it provides a distilled hint to the user
     /// of how downstream code would break with this change.
     #[serde(default)]
```

Comment:
We might not need `#[serde(default)]` here since this should always be specified (given the outer option wasn't `None`) I think, right?